src/yadism/coefficient_functions/coupling_constants.py

```diff
@@ -270,6 +270,43 @@ def get_weight(self, pid, Q2, quark_coupling_type, cc_mask=None):
             return w_phph + w_phZ + w_ZZ
         raise ValueError(f"Unknown process: {self.obs_config['process']}")
 
+    def average_partonic_coupling(self, pid):
+        """Return the linear average of the partonic coupling. Defined only for |NC|.
+
+        Parameters
+        ----------
+        pid : int
+            particle identifier
+
+        Returns
+        -------
+        float
+            weight
+        """
+        if self.obs_config["process"] == "CC":
+            return 0.0
+        # NC or EM
+        # are we in positivity mode?
+        if (
+            self.obs_config["nc_pos_charge"] is not None
+            and self.obs_config["nc_pos_charge"] != "all"
+        ):
+            pos = self.obs_config["nc_pos_charge"][0]
+            pos_pid = 1 + br.quark_names.index(pos)
+            if abs(pid) != pos_pid:
+                return 0.0
+
+        w_ph = self.electric_charge[pid]
+        # pure EM charge
+        if self.obs_config["process"] == "EM":
+            return w_ph
+        # allow Z to be mixed in
+        if self.obs_config["process"] == "NC":
+            w_ZV = self.vectorial_coupling(pid)
+            w_ZA = self.weak_isospin_3[pid]
+            return w_ph + w_ZV + w_ZA
```

Comment:
Did we agreed on sum right, not average?? @andreab1997 @felixhekhorn

Reply:
actually, I wonder about the axial coupling ... a diagram with a single $\gamma_5$ can not survive - so maybe only the vectorial part here ...

Reply:
Should we keep only the vectorial part then?

Reply:
> actually, I wonder about the axial coupling ... a diagram with a single γ5 can not survive - so maybe only the vectorial part here ...

Which is the diagram we're talking about?

Reply:
[Fig1 and Table2 of this paper](https://arxiv.org/pdf/hep-ph/0504242.pdf)

Reply:
Ok, sorry, you're right. I was thinking to the cross-section.

But if we're not considering the whole cross-section we can't rule out the axial contribution. Indeed, it might contribute to the `AA` cross-section. The only thing you can really rule out is the `VA` cross-section, because it would have an odd number of $\gamma_5$ in a trace. But if you are not talking about a cross-section, there is no trace yet, so you can not make considerations about $\gamma_5$ being even or odd, since you have to consider the complex conjugate as well.

If this is at linear level, the only thing you can do is to split the vectorial and axial component.

Reply:
@giacomomagni this discussion only concers `fl11`, right?

Reply:
> @giacomomagni this discussion only concers `fl11`, right?

In that case, @felixhekhorn is right: the axial coupling should not be averaged, since there it would correspond to a loop with a single $\gamma_5$ (just one EW boson per quark-loop).

Reply:
> @giacomomagni this discussion only concers `fl11`, right?

This affects all `fl11, flg` and `flps`.

Reply:
i.e. with the paper notation: `fl11` which couples with the "QED singlet" and thus in Vogt language has two components "non-singlet" and "singlet" (meaning pure-singlet) and `fl11g` which is the gluonic counterpart, right?

all in all: in concerns all diagrams with 2 distinct fermion loops, with the boson coupling only to a single loop - now the vanishing of a single $\gamma_5$ terms should apply to each fermion loop separately and so to all 3 channels

Reply:
All good :)